Add new property to Carousel template and Buttons template

ImJCTW opened the pull request:
Add new property to these two template: 
* Carousel template
  - imageBackgroundColor
  - defaultAction

I am not sure should I move ButtonsTemplate.IsValidColor to Helper class or not.
So I just copy it into  #CarouselColumn.

https://developers.line.biz/en/reference/messaging-api/#carousel

* Buttons template
  - defaultAction

https://developers.line.biz/en/reference/messaging-api/#buttons

dlemstra commented on src/LineBot/Messages/Template/Carousel/CarouselColumn.cs:
```diff
@@ -27,6 +27,8 @@ public sealed class CarouselColumn
         private Uri _thumbnailUrl;
         private string _title;
         private string _text;
+        private string _color = "#FFFFFF";
```
I wonder if we should change the code to remove `null` values when serializing the object. We don't need this default then. Could you take a look at that?

ImJCTW replied:
Agree, I will check these attribute.

dlemstra commented on src/LineBot/Messages/Template/Carousel/CarouselColumn.cs:
```diff
@@ -154,5 +202,29 @@ internal void Validate()
+
+        private static bool IsValidColor(string value)
+        {
+            for (int i = 1; i < value.Length; i++)
```
I do think this should be moved to some helper class. Maybe an extension method for string?

ImJCTW replied:
I will make an extension class for this.

ImJCTW commented:
Sorry for late response, I did not receive notify email.
I am a little busy now, but I will finish these tasks in the near future.
I will available during Chinese New Year at early February.

dlemstra commented:
No problem and thanks for coming back 👍 Feel free to take your time.

ImJCTW commented:
Sorry for delay so long.
 I moved color validator to StringExtension, and add two tests for DefaultAction. 
For ignore null value when output JSON, I added parameters to JsonConvert.SerializeObject and modified tests.
let me know if any changes need to be made. :smile:
